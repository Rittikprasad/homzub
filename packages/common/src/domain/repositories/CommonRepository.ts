import { BootstrapAppService } from '@homzhub/common/src/services/BootstrapAppService';
import { ObjectMapper } from '@homzhub/common/src/utils/ObjectMapper';
import { CarpetArea } from '@homzhub/common/src/domain/models/CarpetArea';
import { Country } from '@homzhub/common/src/domain/models/Country';
import { Currency } from '@homzhub/common/src/domain/models/Currency';
import { DeepLink } from '@homzhub/common/src/domain/models/DeepLink';
import { MarketTrends } from '@homzhub/common/src/domain/models/MarketTrends';
import { OnBoarding } from '@homzhub/common/src/domain/models/OnBoarding';
import { Pillar, PillarTypes } from '@homzhub/common/src/domain/models/Pillar';
import { SocialAuthProvider } from '@homzhub/common/src/domain/models/SocialAuthProvider';
import { Unit } from '@homzhub/common/src/domain/models/Unit';
import { CaseLog } from '@homzhub/common/src/domain/models/CaseLog';
import { User } from '@homzhub/common/src/domain/models/User';
import { IApiClient } from '@homzhub/common/src/network/Interfaces';
import {
  IMarketTrendParams,
  ISubscribeToNewsletterPayload,
  ISupportPayload,
  IDeviceTokenPayload,
  ILimitedOfferPayload,
  IDeepLinkBody,
} from '@homzhub/common/src/domain/repositories/interfaces';
import { SocialAuthProviders } from '@homzhub/common/src/constants/SocialAuthProviders';

const ENDPOINTS = {
  getCountryCodes: 'v1/countries/',
  getCurrencyCodes: 'v1/currency-codes/',
  carpetAreaUnits: 'v1/carpet-area-units/',
  maintenanceUnits: 'v1/list-of-values/maintenance-units/',
  onBoarding: 'v3/onboardings/',
  supportCategories: 'v1/list-of-values/client-support-categories/',
  supportContact: 'v1/client-support/contacts/',
  clientSupport: 'v1/client-support/',
  getMarketTrends: 'v1/market-trends/',
  getPillars: 'v1/pillars/',
  subscribeToNewsLetter: 'v1/newslettters/subscriptions/',
  updateMarketTrends: (id: number): string => `v1/market-trends/${id}/`,
  postDeviceToken: (): string => 'v1/devices/',
  deepLinks: 'v1/deep-links',
};

class CommonRepository {
  private apiClient: IApiClient;

  public constructor() {
    this.apiClient = BootstrapAppService.clientInstance;
  }

  public getCountryCodes = async (): Promise<Country[]> => {
    const response = await this.apiClient.get(ENDPOINTS.getCountryCodes);
    return ObjectMapper.deserializeArray(Country, response);
  };

  public getCurrencyCodes = async (): Promise<Currency[]> => {
    const response = await this.apiClient.get(ENDPOINTS.getCurrencyCodes);
    return ObjectMapper.deserializeArray(Currency, response);
  };

  public getCarpetAreaUnits = async (): Promise<CarpetArea[]> => {
    const response = await this.apiClient.get(ENDPOINTS.carpetAreaUnits);
    return ObjectMapper.deserializeArray(CarpetArea, response);
  };

  public getOnBoarding = async (): Promise<OnBoarding[]> => {
    const response = await this.apiClient.get(ENDPOINTS.onBoarding);
    return ObjectMapper.deserializeArray(OnBoarding, response);
  };

  public getSocialMedia = (): SocialAuthProvider[] => {
    return ObjectMapper.deserializeArray(SocialAuthProvider, SocialAuthProviders);
  };

  public getMaintenanceUnits = async (): Promise<Unit[]> => {
    const response = await this.apiClient.get(ENDPOINTS.maintenanceUnits);
    return ObjectMapper.deserializeArray(Unit, response);
  };

  public getSupportCategories = async (): Promise<Unit[]> => {
    const response = await this.apiClient.get(ENDPOINTS.supportCategories);
    return ObjectMapper.deserializeArray(Unit, response);
  };

  public getSupportContacts = async (): Promise<User> => {
    const response = await this.apiClient.get(ENDPOINTS.supportContact);
    return ObjectMapper.deserialize(User, response);
  };

  public getClientSupport = async (): Promise<CaseLog[]> => {
    const response = await this.apiClient.get(ENDPOINTS.clientSupport);
    return ObjectMapper.deserializeArray(CaseLog, response);
  };

  public subscribeToNewsLetter = async (
    payload: ISubscribeToNewsletterPayload | ILimitedOfferPayload
  ): Promise<void> => {
    return await this.apiClient.post(ENDPOINTS.subscribeToNewsLetter, payload);
  };

  public postClientSupport = async (payload: ISupportPayload): Promise<void> => {
    return await this.apiClient.post(ENDPOINTS.clientSupport, payload);
  };

  public getMarketTrends = async (params: IMarketTrendParams): Promise<MarketTrends> => {
    const response = await this.apiClient.get(ENDPOINTS.getMarketTrends, params);
    return ObjectMapper.deserialize(MarketTrends, response);
  };

  public updateMarketTrends = async (id: number): Promise<void> => {
    await this.apiClient.patch(ENDPOINTS.updateMarketTrends(id));
  };

  public getPillars = async (reviewType: PillarTypes): Promise<Pillar[]> => {
    const response = await this.apiClient.get(ENDPOINTS.getPillars, {
      review_type: reviewType,
    });
    return ObjectMapper.deserializeArray(Pillar, response);
  };

  public postDeviceToken = async (requestBody: IDeviceTokenPayload): Promise<void> => {
    return await this.apiClient.post(ENDPOINTS.postDeviceToken(), requestBody);
  };

  public getDeepLink = async (requestBody: IDeepLinkBody): Promise<DeepLink> => {
    const response = await this.apiClient.post(ENDPOINTS.deepLinks, requestBody);
    return ObjectMapper.deserialize(DeepLink, response);
  };
}

const commonRepository = new CommonRepository();
export { commonRepository as CommonRepository };
