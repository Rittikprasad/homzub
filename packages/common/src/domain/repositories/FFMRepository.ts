import { ObjectMapper } from '@homzhub/common/src/utils/ObjectMapper';
import { BootstrapAppService } from '@homzhub/common/src/services/BootstrapAppService';
import { AssetSearch } from '@homzhub/common/src/domain/models/AssetSearch';
import { Feedback } from '@homzhub/common/src/domain/models/Feedback';
import { FFMMetrics } from '@homzhub/common/src/domain/models/FFMMetrics';
import { FFMTicket } from '@homzhub/common/src/domain/models/FFMTicket';
import { FFMVisit } from '@homzhub/common/src/domain/models/FFMVisit';
import { InspectionFinalReport } from '@homzhub/common/src/domain/models/InspectionFinalReport';
import { InspectionReport } from '@homzhub/common/src/domain/models/InspectionReport';
import { OnBoarding } from '@homzhub/common/src/domain/models/OnBoarding';
import { OutsetCheck } from '@homzhub/common/src/domain/models/OutsetCheck';
import { ReportSpace } from '@homzhub/common/src/domain/models/ReportSpace';
import { SpaceDetail } from '@homzhub/common/src/domain/models/SpaceDetail';
import { Ticket } from '@homzhub/common/src/domain/models/Ticket';
import { TicketAction } from '@homzhub/common/src/domain/models/TicketAction';
import { TicketManagement } from '@homzhub/common/src/domain/models/TicketManagement';
import { Unit } from '@homzhub/common/src/domain/models/Unit';
import { IApiClient } from '@homzhub/common/src/network/Interfaces';
import {
  IFFMVisitParam,
  IGetFeedbackParam,
  IGetSpaceDetail,
  IGetTicket,
  IOutsetCheckParam,
  IPostFeedback,
  IReportSpaceParam,
  IUpdateReport,
  IUpdateSpaceParam,
  IUpdateTicket,
  UserRole,
} from '@homzhub/common/src/domain/repositories/interfaces';
import { Tabs } from '@homzhub/common/src/constants/Tabs';

const ENDPOINTS = {
  onBoarding: 'v1/ffm-onboardings',
  roles: 'v1/roles',
  visits: 'v1/ffm/tasks/site-visits/',
  userRole: () => 'v1/ffm/users/',
  visitDetail: (visitId: number): string => `v1/ffm/tasks/site-visits/${visitId}/`,
  rejectReason: (visitId: number): string => `v1/ffm/listing-visits/${visitId}/prospect-feedbacks/reject-reasons/`,
  feedback: (visitId: number): string => `v1/ffm/listing-visits/${visitId}/prospect-feedbacks/`,
  feedbackById: (visitId: number, feedbackId: number): string =>
    `v1/ffm/listing-visits/${visitId}/prospect-feedbacks/${feedbackId}`,
  inspectionReport: 'v1/ffm/tasks/inspection-reports/',
  inspectionReportById: (reportId: number): string => `v1/ffm/tasks/inspection-reports/${reportId}/`,
  outsetsCheck: (reportId: number): string => `v1/ffm/tasks/inspection-reports/${reportId}/outsets/`,
  reportSpaces: (reportId: number): string => `v1/ffm/tasks/inspection-reports/${reportId}/inspection-report-spaces/`,
  reportSpacesDetail: (reportId: number, spaceId: number): string =>
    `v1/ffm/tasks/inspection-reports/${reportId}/inspection-report-spaces/${spaceId}/`,
  spaceInspection: (reportId: number, spaceId: number): string =>
    `v1/ffm/tasks/inspection-reports/${reportId}/inspection-report-spaces/${spaceId}/space-inspections/`,
  finalReport: (reportId: number): string => `v1/ffm/tasks/inspection-reports/${reportId}/final-report/`,
  hotLeaseProperties: 'v1/ffm/assets/lease-listing-hot-properties/',
  hotSaleProperties: 'v1/ffm/assets/sale-listing-hot-properties/',
  managementTab: 'v1/ffm/management-tab',
  tickets: 'v1/ffm/tasks/tickets/',
  ticketManagement: 'v1/ffm/tasks/tickets/management-tab/',
  ticketById: (id: number): string => `v1/ffm/tasks/tickets/${id}/`,
  ticketActions: (id: number): string => `v1/ffm/tasks/tickets/${id}/actions/`,
};

class FFMRepository {
  private apiClient: IApiClient;

  public constructor() {
    this.apiClient = BootstrapAppService.clientInstance;
  }

  public getOnBoarding = async (): Promise<OnBoarding[]> => {
    const response = await this.apiClient.get(ENDPOINTS.onBoarding);
    return ObjectMapper.deserializeArray(OnBoarding, response);
  };

  public getRoles = async (): Promise<Unit[]> => {
    const response = await this.apiClient.get(ENDPOINTS.roles);
    return ObjectMapper.deserializeArray(Unit, response);
  };

  public getVisits = async (param?: IFFMVisitParam): Promise<FFMVisit[]> => {
    const response = await this.apiClient.get(ENDPOINTS.visits, param);
    return ObjectMapper.deserializeArray(FFMVisit, response);
  };

  public getVisitDetail = async (visitId: number): Promise<FFMVisit> => {
    const response = await this.apiClient.get(ENDPOINTS.visitDetail(visitId));
    return ObjectMapper.deserialize(FFMVisit, response);
  };

  public getRejectReason = async (visitId: number): Promise<Unit[]> => {
    const response = await this.apiClient.get(ENDPOINTS.rejectReason(visitId));
    return ObjectMapper.deserializeArray(Unit, response);
  };

  public postFeedback = async (payload: IPostFeedback): Promise<void> => {
    const { visitId, data } = payload;
    return await this.apiClient.post(ENDPOINTS.feedback(visitId), data);
  };

  public getFeedbackById = async (payload: IGetFeedbackParam): Promise<Feedback> => {
    const response = await this.apiClient.get(ENDPOINTS.feedbackById(payload.visitId, payload.feedbackId));
    return ObjectMapper.deserialize(Feedback, response);
  };

  public getInspectionReport = async (payload: string): Promise<InspectionReport> => {
    const response = await this.apiClient.get(ENDPOINTS.inspectionReport, { status_category: payload });
    return ObjectMapper.deserialize(InspectionReport, response);
  };

  public updateInspectionReport = async (payload: IUpdateReport): Promise<void> => {
    return await this.apiClient.patch(ENDPOINTS.inspectionReportById(payload.reportId), {
      status: payload.status,
    });
  };

  public outsetsCheck = async (payload: IOutsetCheckParam): Promise<OutsetCheck> => {
    const response = await this.apiClient.post(ENDPOINTS.outsetsCheck(payload.reportId), payload.body);
    return ObjectMapper.deserialize(OutsetCheck, response);
  };

  public getReportSpaces = async (payload: number): Promise<ReportSpace[]> => {
    const response = await this.apiClient.get(ENDPOINTS.reportSpaces(payload));
    return ObjectMapper.deserializeArray(ReportSpace, response);
  };

  public createReportSpaces = async (payload: IReportSpaceParam): Promise<void> => {
    return await this.apiClient.post(ENDPOINTS.reportSpaces(payload.reportId), payload.body);
  };

  public updateSpaceDetail = async (payload: IUpdateSpaceParam): Promise<void> => {
    return await this.apiClient.put(ENDPOINTS.spaceInspection(payload.reportId, payload.spaceId), payload.body);
  };

  public getSpaceDetail = async (payload: IGetSpaceDetail): Promise<SpaceDetail> => {
    const response = await this.apiClient.get(ENDPOINTS.reportSpacesDetail(payload.reportId, payload.spaceId));
    return ObjectMapper.deserialize(SpaceDetail, response);
  };

  public getFinalReport = async (payload: number): Promise<InspectionFinalReport> => {
    const response = await this.apiClient.get(ENDPOINTS.finalReport(payload));
    return ObjectMapper.deserialize(InspectionFinalReport, response);
  };

  public getHotProperties = async (payload: Tabs): Promise<AssetSearch> => {
    const url = payload === Tabs.RENT ? ENDPOINTS.hotLeaseProperties : ENDPOINTS.hotSaleProperties;
    const response = await this.apiClient.get(url);
    return ObjectMapper.deserialize(AssetSearch, response);
  };

  public getManagementTab = async (): Promise<FFMMetrics> => {
    const response = await this.apiClient.get(ENDPOINTS.managementTab);
    return ObjectMapper.deserialize(FFMMetrics, response);
  };

  public getTickets = async (payload: IGetTicket): Promise<FFMTicket> => {
    const response = await this.apiClient.get(ENDPOINTS.tickets, payload);
    return ObjectMapper.deserialize(FFMTicket, response);
  };

  public updateTicket = async (payload: IUpdateTicket): Promise<void> => {
    return await this.apiClient.patch(ENDPOINTS.ticketById(payload.id), { action: payload.action });
  };

  public getTicketDetail = async (payload: number): Promise<Ticket> => {
    const response = await this.apiClient.get(ENDPOINTS.ticketById(payload));
    return ObjectMapper.deserialize(Ticket, response);
  };

  public getTicketActions = async (ticketId: number): Promise<TicketAction[]> => {
    const response = await this.apiClient.get(ENDPOINTS.ticketActions(ticketId));
    return ObjectMapper.deserializeArray(TicketAction, response);
  };

  public getTicketManagement = async (): Promise<TicketManagement> => {
    const response = await this.apiClient.get(ENDPOINTS.ticketManagement);
    return ObjectMapper.deserialize(TicketManagement, response);
  };

  public updateUserRole = async (payload: UserRole): Promise<void> => {
    const { data } = payload;
    return await this.apiClient.post(ENDPOINTS.userRole(), data);
  };
}

const ffmRepository = new FFMRepository();
export { ffmRepository as FFMRepository };
