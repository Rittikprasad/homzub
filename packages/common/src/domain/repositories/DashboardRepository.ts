import { BootstrapAppService } from '@homzhub/common/src/services/BootstrapAppService';
import { ObjectMapper } from '@homzhub/common/src/utils/ObjectMapper';
import { IApiClient } from '@homzhub/common/src/network/Interfaces';
import { INotificationsPayload } from '@homzhub/common/src/domain/repositories/interfaces';
import { AssetMetrics } from '@homzhub/common/src/domain/models/AssetMetrics';
import { AssetAdvertisement } from '@homzhub/common/src/domain/models/AssetAdvertisement';
import { AssetNotifications } from '@homzhub/common/src/domain/models/AssetNotifications';

const ENDPOINTS = {
  getAssetMetrics: (version: string): string => `${version}/dashboard/management-tab/`,
  getAdvertisements: (): string => 'v1/advertisements/',
  getGeneralLedgers: (): string => 'v1/general-ledgers/overall-performances/',
  getAssetNotifications: (): string => 'v1/notifications/',
  updateNotifications: (id: number): string => `v1/notifications/${id}/`,
  markAllNotificationsRead: (): string => 'v1/notifications/mark-as-read/',
};

class DashboardRepository {
  private apiClient: IApiClient;

  public constructor() {
    this.apiClient = BootstrapAppService.clientInstance;
  }

  // Todo (Praharsh) : Remove argument when web dashboard is updated with v3.
  public getAssetMetrics = async (version = 'v2'): Promise<AssetMetrics> => {
    const response = await this.apiClient.get(ENDPOINTS.getAssetMetrics(version));
    return ObjectMapper.deserialize(AssetMetrics, response);
  };

  public getAdvertisements = async (payload?: { category: string }): Promise<AssetAdvertisement> => {
    const response = await this.apiClient.get(ENDPOINTS.getAdvertisements(), payload || {});
    return ObjectMapper.deserialize(AssetAdvertisement, response);
  };

  public getAssetNotifications = async (requestPayload: INotificationsPayload): Promise<AssetNotifications> => {
    const response = await this.apiClient.get(ENDPOINTS.getAssetNotifications(), requestPayload);
    return ObjectMapper.deserialize(AssetNotifications, response);
  };

  public updateNotificationStatus = async (id: number): Promise<void> => {
    return await this.apiClient.patch(ENDPOINTS.updateNotifications(id));
  };

  public markAllNotificationsRead = async (latestCreatedAt: string): Promise<void> => {
    return await this.apiClient.post(ENDPOINTS.markAllNotificationsRead(), { created_at: latestCreatedAt });
  };
}

const dashboardRepository = new DashboardRepository();
export { dashboardRepository as DashboardRepository };
