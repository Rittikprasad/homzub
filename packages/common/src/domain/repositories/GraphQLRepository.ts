import axios from 'axios';
import { IFAQCategory, IFAQs } from '@homzhub/common/src/domain/repositories/interfaces';

const ENDPOINTS = {
  HOMZHUB: 'https://api-ap-northeast-1.graphcms.com/v2/ckiqymr7djhfb01z1di6m4fjs/master',
  FAQ: 'https://api-ap-northeast-1.graphcms.com/v2/ckmbpffvbztly01xw7dsudj1u/master',
};

export interface IFeaturedProperties {
  address: string;
  category: string;
  coverImage: { url: string };
  developer: { name: string };
  featured: boolean;
  id: string;
  images: { url: string }[];
  possessionDate: string;
  priceRange: string;
  projectName: string;
  slug: string;
  stage: string;
  typesAvailable: string;
}

const getQuery = (queryKey: string, param?: string): string => {
  switch (queryKey) {
    case 'featuredProperties':
      return `query {
        properties(orderBy: updatedAt_ASC) {
            id
            address
            possessionDate
            category
            priceRange
            typesAvailable
            coverImage {
                url
            }
            developer {
                name
            }
            projectName
            slug
            stage
            featured
            images {
                url
            }
        }
    }`;
    case 'faqAllCategories':
      return `query MyQuery {
      categories {
        activeIcon {
          url
        }  
        inActiveIcon {
          url
        }
        id
        title
      }
    }`;
    case 'faqAllQuestions':
      return `query MyQuery {
      faqs() {
        question
        answerRichText{html}
        category{id title}
      }
    }`;
    case 'faqSearchQuestions':
      return `query MyQuery {
      faqs(where: {question_contains: "${param}"}) {
        question
        answerRichText{html}
        category{id title}
      }
    }`;
    case 'faqCategoryQuestions':
      return `query MyQuery {
    faqs(where: {category: {AND: {id: "${param}"}}}) {
      question
      answerRichText{html}
      category{id title}
    }
  }`;
    default:
      return '';
  }
};
class GraphQLRepository {
  public getFeaturedProperties = async (): Promise<IFeaturedProperties[]> => {
    const response = await axios({
      url: ENDPOINTS.HOMZHUB,
      method: 'post',
      data: {
        query: getQuery('featuredProperties'),
      },
    }).then((result) => {
      return result.data;
    });
    return response.data.properties;
  };

  public getFAQAllCategories = async (): Promise<IFAQCategory[]> => {
    const response = await axios({
      url: ENDPOINTS.FAQ,
      method: 'post',
      data: {
        query: getQuery('faqAllCategories'),
      },
    }).then((result) => {
      return result.data;
    });
    return response.data.categories;
  };

  public getFAQAllQuestions = async (): Promise<IFAQs[]> => {
    const response = await axios({
      url: ENDPOINTS.FAQ,
      method: 'post',
      data: {
        query: getQuery('faqAllQuestions'),
      },
    }).then((result) => {
      return result.data;
    });
    return response.data.faqs;
  };

  public getFAQSearchQuestions = async (param: string): Promise<IFAQs[]> => {
    const response = await axios({
      url: ENDPOINTS.FAQ,
      method: 'post',
      data: {
        query: getQuery('faqSearchQuestions', param),
      },
    }).then((result) => {
      return result.data;
    });
    return response.data.faqs;
  };

  public getFAQCategoryQuestions = async (param: string): Promise<IFAQs[]> => {
    const response = await axios({
      url: ENDPOINTS.FAQ,
      method: 'post',
      data: {
        query: getQuery('faqCategoryQuestions', param),
      },
    }).then((result) => {
      return result.data;
    });
    return response.data.faqs;
  };
}

const graphQLRepository = new GraphQLRepository();
export { graphQLRepository as GraphQLRepository };
