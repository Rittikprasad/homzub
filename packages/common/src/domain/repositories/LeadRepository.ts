import { BootstrapAppService } from '@homzhub/common/src/services/BootstrapAppService';
import { IApiClient } from '@homzhub/common/src/network/Interfaces';
import { ILeadPayload } from '@homzhub/common/src/domain/repositories/interfaces';

const ENDPOINTS = {
  postLeaseLead: (propertyTermId: number): string => `v1/lease-listings/${propertyTermId}/leads/`,
  postSaleLead: (propertyTermId: number): string => `v1/sale-listings/${propertyTermId}/leads/`,
};

class LeadRepository {
  private apiClient: IApiClient;

  public constructor() {
    this.apiClient = BootstrapAppService.clientInstance;
  }

  // TODO: Need to verify return body and type after signup flow
  public postLeaseLeadDetail = async (payload: ILeadPayload): Promise<any> => {
    const { propertyTermId, data } = payload;
    return await this.apiClient.post(ENDPOINTS.postLeaseLead(propertyTermId), data);
  };

  // TODO: Need to verify return body and type after signup flow
  public postSaleLeadDetail = async (payload: ILeadPayload): Promise<any> => {
    const { propertyTermId, data } = payload;
    return await this.apiClient.post(ENDPOINTS.postSaleLead(propertyTermId), data);
  };
}

const leadRepository = new LeadRepository();
export { leadRepository as LeadRepository };
