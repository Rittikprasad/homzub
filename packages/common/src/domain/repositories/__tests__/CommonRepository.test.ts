import { BootstrapAppService } from '@homzhub/common/src/services/BootstrapAppService';
import { CommonRepository } from '@homzhub/common/src/domain/repositories/CommonRepository';
import { countryCodes, carpetAreaUnits, currencyCodes } from '@homzhub/common/src/mocks/CommonRepositoryMocks';
import { OnboardingData } from '@homzhub/common/src/mocks/Onboarding';
import { SocialMediaData } from '@homzhub/common/src/mocks/SocialMedia';

jest.mock('@homzhub/common/src/services/storage/StorageService', () => 'StorageService');
jest.mock('@react-native-community/google-signin', () => {});

describe.skip('CommonRepository', () => {
  afterAll(() => {
    jest.resetAllMocks();
  });

  it('should fetch a list of county codes', async () => {
    jest.spyOn(BootstrapAppService.clientInstance, 'get').mockImplementation(() => Promise.resolve(countryCodes));
    const response = await CommonRepository.getCountryCodes();
    expect(response).toMatchSnapshot();
  });

  it('should fetch a list of unit areas', async () => {
    jest.spyOn(BootstrapAppService.clientInstance, 'get').mockImplementation(() => Promise.resolve(carpetAreaUnits));
    const response = await CommonRepository.getCarpetAreaUnits();
    expect(response).toMatchSnapshot();
  });

  it('should fetch a list of currency codes', async () => {
    jest.spyOn(BootstrapAppService.clientInstance, 'get').mockImplementation(() => Promise.resolve(currencyCodes));
    const response = await CommonRepository.getCurrencyCodes();
    expect(response).toMatchSnapshot();
  });

  it('should fetch social media data', () => {
    jest.spyOn(BootstrapAppService.clientInstance, 'get').mockImplementation(() => Promise.resolve(SocialMediaData));
    const response = CommonRepository.getSocialMedia();
    expect(response).toMatchSnapshot();
  });

  it('should fetch OnBoarding screen data', async () => {
    jest.spyOn(BootstrapAppService.clientInstance, 'get').mockImplementation(() => Promise.resolve(OnboardingData));
    const response = await CommonRepository.getOnBoarding();
    expect(response).toMatchSnapshot();
  });

  ['getCountryCodes', 'getCarpetAreaUnits', 'getOnboarding', 'getCurrencyCodes', 'getSocialMedia'].forEach(
    (api: string) => {
      it(`should throw an error in case of ${api} API failure`, async () => {
        jest.spyOn(BootstrapAppService.clientInstance, 'get').mockImplementation(() => {
          throw new Error('Test Error');
        });
        try {
          // @ts-ignore
          await CommonRepository[api]();
        } catch (e) {
          expect(e).toBeTruthy();
        }
      });
    }
  );
});
