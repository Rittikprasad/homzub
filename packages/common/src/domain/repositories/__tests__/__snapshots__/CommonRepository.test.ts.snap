// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonRepository should fetch OnBoarding screen data 1`] = `
Array [
  OnBoarding {
    "_description": "Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a.",
    "_imageUrl": "https://homzhub-bucket.s3.amazonaws.com/Search_Onboarding.svg",
    "_title": "Search",
  },
  OnBoarding {
    "_description": "Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a.",
    "_imageUrl": "https://homzhub-bucket.s3.amazonaws.com/Post_Onboarding.svg",
    "_title": "Post",
  },
  OnBoarding {
    "_description": "Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a.",
    "_imageUrl": "https://homzhub-bucket.s3.amazonaws.com/Maintain_Onboarding.svg",
    "_title": "Maintain",
  },
]
`;

exports[`CommonRepository should fetch a list of county codes 1`] = `
Array [
  CountryCode {
    "_alphaCode": "IN",
    "_country": "INDIA",
    "_countryCode": "+91",
    "_id": 1,
  },
  CountryCode {
    "_alphaCode": "SG",
    "_country": "SINGAPORE",
    "_countryCode": "+65",
    "_id": 2,
  },
  CountryCode {
    "_alphaCode": "AU",
    "_country": "AUSTRALIA",
    "_countryCode": "+36",
    "_id": 3,
  },
]
`;

exports[`CommonRepository should fetch a list of currency codes 1`] = `
Array [
  Currency {
    "_currencyCode": "AUD",
    "_currencyName": "AUD",
    "_currencySymbol": "$",
  },
  Currency {
    "_currencyCode": "INR",
    "_currencyName": "AUD",
    "_currencySymbol": "â‚¹",
  },
  Currency {
    "_currencyCode": "SGD",
    "_currencyName": "AUD",
    "_currencySymbol": "$",
  },
]
`;

exports[`CommonRepository should fetch a list of unit areas 1`] = `
Array [
  CarpetArea {
    "_id": 1,
    "_label": "Square Feet",
    "_name": "SQ_FT",
    "_title": "Sq.ft",
  },
  CarpetArea {
    "_id": 2,
    "_label": "Square Meter",
    "_name": "SQ_M",
    "_title": "Sq.m",
  },
  CarpetArea {
    "_id": 3,
    "_label": "Square Yard",
    "_name": "SQ_YARD",
    "_title": "Sq.yard",
  },
  CarpetArea {
    "_id": 4,
    "_label": "Acres",
    "_name": "ACRES",
    "_title": "Acres",
  },
  CarpetArea {
    "_id": 5,
    "_label": "Hectares",
    "_name": "HECTARES",
    "_title": "Hectares",
  },
]
`;

exports[`CommonRepository should fetch social media data 1`] = `
Array [
  SocialAuthProvider {
    "_clientID": "1008014324415-kclkc41epc41nqjp0uc4gdgg528m1nc6.apps.googleusercontent.com",
    "_description": "with Google",
    "_provider": "GOOGLE",
  },
  SocialAuthProvider {
    "_clientID": "",
    "_description": "with Facebook",
    "_provider": "FACEBOOK",
  },
  SocialAuthProvider {
    "_clientID": "",
    "_description": "with Linkedin",
    "_provider": "LINKEDIN",
  },
]
`;
